src/section/section_keys.rs

```diff
@@ -50,12 +65,92 @@ impl SectionKeysProvider {
 
     pub fn finalise_dkg(&mut self, public_key: &bls::PublicKey) {
         if let Some(share) = &self.pending {
-            let pending_public_key = share.public_key_set.public_key();
+            if *public_key != share.public_key_set.public_key() {
+                return;
+            }
+        }
+        if let Some(share) = self.pending.take() {
+            if let Some(evicted) = self.cache.add(public_key, share) {
+                trace!("evicted old key from cache: {:?}", evicted);
+            }
+            trace!("finalised DKG: {:?}", public_key);
+        }
+    }
+}
+
+/// Implementation of simple cache.
+#[derive(Debug)]
+pub struct KeyCache {
+    map: BTreeMap<Vec<u8>, SectionKeyShare>,
+    list: VecDeque<bls::PublicKey>,
+    capacity: usize,
```

Comment:
Why do we need a separate `capacity` here? `VecDeque` already has a capacity.

Reply:
Think it was just left by mistake from before the vecdequeue was used. :+1: Removing it.